pkg/resource/statefulset.go

```diff
@@ -148,6 +149,22 @@ func (b StatefulSetBuilder) Placeholder() runtime.Object {
 	}
 }
 
+func (b StatefulSetBuilder) SetAnnotations(obj runtime.Object) error {
+	ss, ok := obj.(*appsv1.StatefulSet)
+	if !ok {
+		return errors.New("failed to cast to StatefulSet object")
+	}
+	ss.Annotations[CrdbVersion] = b.Cluster.Status().Version
```

Comment:
check that Version is not empty

Reply:
this method is called from sandbox.go and will never be empty:
```go
	aa := u.GetAnnotations()
	if aa != nil {
		delete(aa, kube.LastAppliedAnnotation)
		if len(aa) > 0 {
			u.SetAnnotations(aa)
		} else {
			unstructured.RemoveNestedField(u.Object, "metadata", "annotations")
		}
	}
```